### PR TITLE
✨  Allow update service version with start view

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

In the context of sub-apps, web and mobile developers want to only see RUM events happening in their area of ownership. For example, a developer working on the Log Management product at Datadog wants to see RUM events that happened in the Log Management only. To do so we will make the service attribute first-class-citizen in RUM and allow customers to update the current service and version.

This PR allow the user to update the service and the version during the lifespan of a session with the startView API

## Changes

Update startView API
Remove service and version tags

## Testing

- [x] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/assembly.spec.ts [hunk 1]
```diff
@@ -47,7 +57,7 @@ describe('rum assembly', () => {
           serverRumEvents.push(serverRumEvent)
         )
         startRumAssembly(
-          configuration,
+          combine(configuration, extraConfigurationOptions),
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/assembly.spec.ts:60]: **💬 suggestion:** ‏In spec files, feel free to use spread:
```suggestion
          {...configuration, ...extraConfigurationOptions},
```

File: packages/rum-core/src/domain/assembly.spec.ts [hunk 2]
```diff
@@ -24,19 +31,22 @@ describe('rum assembly', () => {
   let setupBuilder: TestSetupBuilder
   let commonContext: CommonContext
   let serverRumEvents: RumEvent[]
-
+  let extraConfigurationOptions: object = {}
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/assembly.spec.ts:34]:
```suggestion
  let extraConfigurationOptions: Partial<RumConfiguration> = {}
```

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 3]
```diff
@@ -158,6 +159,14 @@ export function makeRumPublicApi(
     )
   }
 
+  const startView: {
+    (name: string): void
+    // (options: StartViewOptions): void // uncomment when removing the feature flag
+  } = monitor((options?: string) => {
```
[the remark this answers is not in the document]
Comment (amortemousque) [reply packages/rum-core/src/boot/rumPublicApi.ts:165]: With this signature it's not clear what to send as string  `options?: string` that's why I choose the other option

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 4]
```diff
@@ -52,6 +47,12 @@ interface RumPublicApiOptions {
   ignoreInitIfSyntheticsWillInjectRum?: boolean
 }
 
+interface StartViewOptions {
+  name?: string
+  service?: string
+  version?: string
+}
```
[the remark this answers is not in the document]
Comment (amortemousque) [reply packages/rum-core/src/boot/rumPublicApi.ts:54]: Let use the same then ;)

**Pull request merged**